testsuite/Test/Tasty/Clash/NetlistTest.hs

```diff
@@ -0,0 +1,108 @@
+{-# LANGUAGE CPP #-}
+
+#include "MachDeps.h"
+
+-- | This module exposes a function for running the clash compiler as far as
+-- the netlist stage, and returns the netlist AST before it is serialized.
+--
+-- This module is somewhat of a nasty hack (inspired by clash-benchmark) as
+-- clash-lib currently lacks a suitable top-level API for running the compiler
+-- only as far as a required stage.
+--
+module Test.Tasty.Clash.NetlistTest
+  ( runToNetlistStage
+  ) where
+
+import qualified Prelude as P
+import           Clash.Prelude
+
+import           Clash.Annotations.BitRepresentation.Internal
+import           Clash.Backend as Backend
+import           Clash.Backend.VHDL
+import           Clash.Core.Name
+import           Clash.Core.TyCon
+import           Clash.Core.Type
+import           Clash.Core.Var
+import           Clash.Driver as Driver
+import           Clash.Driver.Types
+import           Clash.GHC.Evaluator (reduceConstant)
+import           Clash.GHC.GenerateBindings
+import           Clash.GHC.NetlistTypes
+import           Clash.Netlist
+import           Clash.Netlist.BlackBox.Types (HdlSyn(Other))
+import           Clash.Netlist.Types hiding (backend, hdlDir)
+import           Clash.Util
+
+import           Util
+
+import qualified Control.Concurrent.Supply as Supply
+import           Control.DeepSeq (force)
+import           Control.Monad.State.Strict (State)
+import qualified Control.Monad.State as State
+import           Data.HashMap.Strict (HashMap)
+import qualified Data.HashMap.Strict as HashMap
+import           Data.Maybe
+import qualified Data.Text as Text
+import           System.FilePath ((</>))
+
+typeTrans
+  :: CustomReprs
+  -> TyConMap
+  -> Type
+  -> State HWMap (Maybe (Either String FilteredHWType))
+typeTrans = ghcTypeToHWType WORD_SIZE_IN_BITS True
+
+mkClashOpts :: [FilePath] -> ClashOpts
+mkClashOpts idirs = defClashOpts
+  { opt_cachehdl     = False
+  , opt_errorExtra   = True
+  , opt_floatSupport = True
+  , opt_importPaths  = idirs
+  }
+
+backend :: VHDLState
```

Comment:
Can we test this with the Verilog backend

Reply:
We can, does the choice of backend affect the generated netlist?

Reply:
Yes, but usually minimally. Some blackboxes will have definitions that can be used inline in one language, but would require an assignment declaration in anther.

Reply:
@christiaanb Did you mean "test with Verilog _too_", or would you like to discard the VHDL test altogether?

Reply:
Update: The current version allows running netlist tests against any backend, so I would say this is probably resolved.

Q. Can we test this with the Verilog backend?
A. Yes. Yes we can